### PR TITLE
Use protobufs from submodule

### PR DESCRIPTION
**Pull request by psychonic**
As discussed from time to time, use SteamDB's Protobufs submodule instead of continuing to maintain our own. This makes keeping up to date simpler.

Using
https://github.com/SteamDatabase/Protobufs

Since we can't have individual generate scripts for each app in the folder with their proto files, instead switched to single script to regenerate all, running from the Protogen folder. If that's not desired, we can still do individual bats, just ran from elsewhere. This just removed much of the boilerplate needed.

Also, with this change, the classes for the new Steam chat are using the protos from Steam Client itself, rather than from the friends js.

### CODE REVIEW

**Comment by xPaw** [comment]
I suggest removing `gc.proto` and using the proto from `steam/steammessages_base.proto` instead.

That has all the new fields, except for 200/201 which I suspect are only sent between GC servers anyway.

**Comment by yaakov-h** [comment]
Note to self: close #16 on merge.

File: Resources/Protogen/generate-all.ps1 [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+param([string[]]$ProtoDir)
+
+$ProtoGenDir = $PSScriptRoot
+$ProtoGenExe = Join-Path $PSScriptRoot 'protogen.exe'
+$ProtoBase = Join-Path $PSScriptRoot '..\ProtoBufs'
+$SK2Base = Join-Path $PSScriptRoot '..\..\SteamKit2\SteamKit2\Base\Generated'
+
+Push-Location
+
+$protos = Import-Csv -LiteralPath (Join-Path $PSScriptRoot 'protos.csv') |
```
Comment (yaakov-h) [inline Resources/Protogen/generate-all.ps1]: Neat!

**Comment by yaakov-h** [comment]
Probably something for a later stage, but I'd love to have this automatically submit a PR when the protobufs repo gets updated.

**Comment by xPaw** [comment]
@yaakov-h That proto only exists in CS:GO now: https://github.com/SteamDatabase/Protobufs/search?q=CSOPartyInvite&unscoped_q=CSOPartyInvite

It was generated from tf2 beta before.

**Pull request merged**